### DISCUSSION TITLE
Migrated to Veramo 3.1.0 gave me a query failed

### DISCUSSION THREAD

**Discussion by marytellezm** [decentralized-identity/veramo#880] [Q&A]
When I migrated to Veramo 3.1.0 it gave me a query failed
query failed: INSERT INTO "temporary_key"("kid", "type") SELECT "kid", "type" FROM "key"
error: [Error: SQLITE_CONSTRAINT: NOT NULL constraint failed: temporary_key.kms] {
  errno: 19,
  code: 'SQLITE_CONSTRAINT'
}
QueryFailedError: SQLITE_CONSTRAINT: NOT NULL constraint failed: temporary_key.kms
    at QueryFailedError.TypeORMError [as constructor] (...\node_modules\typeorm\error\TypeORMError.js:9:28)
    at new QueryFailedError ...\node_modules\typeorm\error\QueryFailedError.js:13:28)
    at Statement.handler (...\node_modules\typeorm\driver\sqlite\SqliteQueryRunner.js:79:38) {
  query: 'INSERT INTO "temporary_key"("kid", "type") SELECT "kid", "type" FROM "key"',
  parameters: undefined,
  driverError: [Error: SQLITE_CONSTRAINT: NOT NULL constraint failed: temporary_key.kms] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT'
  },
  errno: 19,
  code: 'SQLITE_CONSTRAINT'
}
Please help

**Comment by mirceanis** [comment]
What version were you using before?

**Reply by marytellezm** [reply]
veramo 1.2.0

**Reply by mirceanis** [reply]
Did you try to go through this migration guide?

https://veramo.io/blog/version_3.0_release

**Reply by marytellezm** [reply]
Yes, i use the migration guide
